syncer/syncer.go

```diff
@@ -1132,580 +1127,627 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
-			)
-
-			// for DDL, we don't apply operator until we try to execute it.
-			// so can handle sharding cases
-			sqls, onlineDDLTableNames, err = s.resolveDDLSQL(parser2, parseResult.stmt, string(ev.Schema))
+	case replication.DELETE_ROWS_EVENTv0, replication.DELETE_ROWS_EVENTv1, replication.DELETE_ROWS_EVENTv2:
+		if !applied {
+			sqls, keys, args, err = genDeleteSQLs(param)
 			if err != nil {
-				log.Infof("[query]%s [last pos]%v [current pos]%v [current gtid set]%v", sql, lastPos, currentPos, ev.GSet)
-				log.Errorf("fail to be parsed, error %v", err)
-				return errors.Trace(err)
+				return errors.Annotatef(err, "gen delete sqls failed, schema: %s, table: %s", table.schema, table.name)
 			}
+		}
+		binlogEvent.WithLabelValues("delete_rows", s.cfg.Name).Observe(time.Since(startTime).Seconds())
+		*latestOp = del
+	}
 
-			if len(onlineDDLTableNames) > 1 {
-				return errors.NotSupportedf("online ddl changes on multiple table: %s", string(ev.Query))
-			}
+	if s.tracer.Enable() {
+		traceEvent, traceErr := s.tracer.CollectSyncerBinlogEvent(traceSource, safeMode.Enable(), tryReSync, *lastPos, *currentPos, int32(header.EventType), int32(*latestOp))
+		if traceErr != nil {
+			log.Infof("[syncer] tracer collect syncer binlog error: %v", errors.ErrorStack(traceErr))
+		}
+		*traceID = traceEvent.Base.TraceID
+	}
 
-			binlogEvent.WithLabelValues("query", s.cfg.Name).Observe(time.Since(startTime).Seconds())
-
-			/*
-				we construct a application transaction for ddl. we save checkpoint after we execute all ddls
-				Here's a brief discussion for implement:
-				* non sharding table: make no difference
-				* sharding table - we limit one ddl event only contains operation for same table
-				  * drop database / drop table / truncate table: we ignore these operations
-				  * create database / create table / create index / drop index / alter table:
-					operation is only for same table,  make no difference
-				  * rename table
-					* online ddl: we would ignore rename ghost table,  make no difference
-					* other rename: we don't allow user to execute more than one rename operation in one ddl event, then it would make no difference
-			*/
-			var (
-				ddlInfo        *shardingDDLInfo
-				needHandleDDLs []string
-				targetTbls     = make(map[string]*filter.Table)
-			)
-			for _, sql := range sqls {
-				sqlDDL, tableNames, stmt, err := s.handleDDL(parser2, string(ev.Schema), sql)
-				if err != nil {
-					return errors.Trace(err)
-				}
-				if len(sqlDDL) == 0 {
-					binlogSkippedEventsTotal.WithLabelValues("query", s.cfg.Name).Inc()
-					log.Warnf("[query-sql]%s [schema]:%s", sql, string(ev.Schema))
-					continue
-				}
+	for i := range sqls {
+		var arg []interface{}
+		var key []string
+		if args != nil {
+			arg = args[i]
+		}
+		if keys != nil {
+			key = keys[i]
+		}
+		err = s.commitJob(insert, originSchema, originTable, table.schema, table.name, sqls[i], arg, key, true, *lastPos, *currentPos, nil, *traceID)
```

Comment:
```suggestion
		err = s.commitJob(*latestOp, originSchema, originTable, table.schema, table.name, sqls[i], arg, key, true, *lastPos, *currentPos, nil, *traceID)
```
?
can add a `default` case to `return`?